File: go/example_code/iam/iam_createaccountalias.go
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package main

import (
	"fmt"
	"os"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/iam"
)

// Usage:
// go run iam_createaccountalias.go <alias>
func main() {
	// Initialize a session in us-west-2 that the SDK will use to load
	// credentials from the shared credentials file ~/.aws/credentials.
	sess, err := session.NewSession(&aws.Config{
		Region: aws.String("us-west-2")},
	)

	// Create a IAM service client.
	svc := iam.New(sess)

	_, err = svc.CreateAccountAlias(&iam.CreateAccountAliasInput{
		AccountAlias: &os.Args[1],
	})

	if err != nil {
		fmt.Println("Error", err)
		return
	}

	fmt.Printf("Account alias %s has been created\n", os.Args[1])
}

File: cpp/example_code/iam/iam_create_account_alias.cpp
/*
   Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.

   This file is licensed under the Apache License, Version 2.0 (the "License").
   You may not use this file except in compliance with the License. A copy of
   the License is located at

    http://aws.amazon.com/apache2.0/

   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
   CONDITIONS OF ANY KIND, either express or implied. See the License for the
   specific language governing permissions and limitations under the License.
*/
#include <aws/core/Aws.h>

#include <aws/iam/IAMClient.h>
#include <aws/iam/model/CreateAccountAliasRequest.h>

#include <iostream>

/**
 * Creates an alias for an AWS account, based on command line input
 */
int main(int argc, char** argv)
{
    if(argc != 2)
    {
        std::cout << "Usage: iam_create_account_alias <alias_name>" << std::endl;
        return 1;
    }

    Aws::String aliasName(argv[1]);

    Aws::SDKOptions options;
    Aws::InitAPI(options);

    {
        Aws::IAM::IAMClient iamClient;

        Aws::IAM::Model::CreateAccountAliasRequest createAccountAliasRequest;
        createAccountAliasRequest.SetAccountAlias(aliasName);

        auto createAccountAliasOutcome = iamClient.CreateAccountAlias(createAccountAliasRequest);
        if (!createAccountAliasOutcome.IsSuccess())
        {
            std::cout << "Error creating account alias " << aliasName << ": " <<
            createAccountAliasOutcome.GetError().GetMessage() << std::endl;
        }
        else
        {
            std::cout << "Successfully created account alias " << aliasName << std::endl;
        }
    }

    Aws::ShutdownAPI(options);

    return 0;
}



